backend/PolicyAdmin/PolicyConverter.cs

```diff
@@ -0,0 +1,269 @@
+                    xacmlAttributeDesignator.MustBePresent = false;
+
+                    XacmlMatch xacmlMatch = new XacmlMatch(new Uri(XacmlConstants.AttributeMatchFunction.StringEqual), xacmlAttributeValue, xacmlAttributeDesignator);
+                    matches.Add(xacmlMatch);
+                }
+
+                XacmlAllOf xacmlAllOf = new XacmlAllOf(matches);
+                resourceAllOfs.Add(xacmlAllOf);
+
+            }
+
+            return new XacmlAnyOf(resourceAllOfs);
+        }
+
+
+        private static XacmlAnyOf GetSubjectAnyOfs(List<string> subjects)
+        {
+            List<XacmlAllOf> subjectAllOfs = new List<XacmlAllOf>();
+            foreach (string subject in subjects)
+            {
+                List<XacmlMatch> matches = new List<XacmlMatch>();
+                int splitLocation = subject.LastIndexOf(":");
+                string attributeDesignator = subject.Substring(0, splitLocation);
+                string attributeValue = subject.Substring(splitLocation + 1);
+
+                XacmlAttributeValue xacmlAttributeValue = new XacmlAttributeValue(new Uri(XacmlConstants.DataTypes.XMLString));
+                xacmlAttributeValue.Value = attributeValue;
+
+                XacmlAttributeDesignator xacmlAttributeDesignator = new XacmlAttributeDesignator(new Uri(attributeDesignator), new Uri(XacmlConstants.DataTypes.XMLString));
+                xacmlAttributeDesignator.Category = new Uri(XacmlConstants.MatchAttributeCategory.Subject);
+                xacmlAttributeDesignator.MustBePresent = false;
+
+                XacmlMatch xacmlMatch = new XacmlMatch(new Uri(XacmlConstants.AttributeMatchFunction.StringEqualIgnoreCase), xacmlAttributeValue, xacmlAttributeDesignator);
+                matches.Add(xacmlMatch);
+
+                XacmlAllOf xacmlAllOf = new XacmlAllOf(matches);
+                subjectAllOfs.Add(xacmlAllOf);
+            }
+
+            return new XacmlAnyOf(subjectAllOfs);
+        }
+
+        private static XacmlAnyOf GetActionAnyOfs(List<string> actions)
+        {
+            List<XacmlAllOf> actionAllOfs = new List<XacmlAllOf>();
+
+            foreach (string action in actions)
+            {
+                List<XacmlMatch> matches = new List<XacmlMatch>();
+                XacmlAttributeValue xacmlAttributeValue = new XacmlAttributeValue(new Uri(XacmlConstants.DataTypes.XMLString));
+                xacmlAttributeValue.Value = action;
+
+                XacmlAttributeDesignator xacmlAttributeDesignator = new XacmlAttributeDesignator(new Uri(XacmlConstants.MatchAttributeIdentifiers.ActionId), new Uri(XacmlConstants.DataTypes.XMLString));
+                xacmlAttributeDesignator.Category = new Uri(XacmlConstants.MatchAttributeCategory.Action);
+                xacmlAttributeDesignator.MustBePresent = false;
+
+                XacmlMatch xacmlMatch = new XacmlMatch(new Uri(XacmlConstants.AttributeMatchFunction.StringEqual), xacmlAttributeValue, xacmlAttributeDesignator);
```

Comment:
For subject the default is StringEqualIgnoreCase but action is StringEqual.
Any reason for this or why it shouldn't have the same default behaviour?

Reply:
This is more historical reasons. But I see your point. I original did not want to mess things up with back forward conversion of app policies